[Flatpak] Update nightly runner script for 23.08 SDK and GTK4 builds

philn opened the pull request:
#### 7cf2d601bd748e9e9de768b04e2f7490d39d5b8f
<pre>
[Flatpak] Update nightly runner script for 23.08 SDK and GTK4 builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=266296">https://bugs.webkit.org/show_bug.cgi?id=266296</a>

Reviewed by Michael Catanzaro.

The --gtk option is now for GTK4 builds and a new --gtk3 option is introduced, to test GTK3 builds.

* Tools/Scripts/webkit-flatpak-run-nightly:
(spinning_cursor):
(MyHTMLParser.handle_starttag):
(get_revision_number):
(run):
(main):
(bwrap_main):

Canonical link: <a href="https://commits.webkit.org/271947@main">https://commits.webkit.org/271947@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f0c79b669c2e214899b06fb81a7b1fbdfb7d1ed0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30142 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8807 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31456 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32645 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27257 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30817 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/11008 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6051 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/27264 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30442 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/7395 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/25709 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6318 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/6474 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/26803 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33982 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/27487 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/27236 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/32647 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6420 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4585 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30452 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8169 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7138 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7173 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6948 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Tools/Scripts/webkit-flatpak-run-nightly:
```diff
@@ -150,8 +153,10 @@ def main(args):
+    platform_group.add_argument('--gtk', action='store_const', dest='platform', const='webkitgtk4',
                                 help='Download and run GTK port build artefacts')
+    platform_group.add_argument('--gtk3', action='store_const', dest='platform', const='webkitgtk',
+                                help='Download and run GTK3 port build artefacts')
```
```suggestion
                                help='Download and run GTK3 port build artifacts')
```

mcatanzaro commented on Tools/Scripts/webkit-flatpak-run-nightly:
```diff
@@ -150,8 +153,10 @@ def main(args):
     platform_group = parser.add_mutually_exclusive_group()
-    platform_group.add_argument('--gtk', action='store_const', dest='platform', const='webkitgtk',
+    platform_group.add_argument('--gtk', action='store_const', dest='platform', const='webkitgtk4',
                                 help='Download and run GTK port build artefacts')
```
Preexisting problem, but might as well fix it:

```suggestion
                                help='Download and run GTK port build artifacts')
```

philn committed:
[Flatpak] Update nightly runner script for 23.08 SDK and GTK4 builds
https://bugs.webkit.org/show_bug.cgi?id=266296

Reviewed by Michael Catanzaro.

The --gtk option is now for GTK4 builds and a new --gtk3 option is introduced, to test GTK3 builds.

* Tools/Scripts/webkit-flatpak-run-nightly:
(spinning_cursor):
(MyHTMLParser.handle_starttag):
(get_revision_number):
(run):
(main):
(bwrap_main):

Canonical link: https://commits.webkit.org/271947@main
